### DISCUSSION TITLE
Removing line code

### DISCUSSION THREAD

**Discussion by siganberg** [dotnet/templating#6785] [Template authoring]
Is there a way in template.json for instructions for removing a line of code if the condition is met? Currently, the only thing I saw is replace which ends up empty line. It works but is not ideal as it looks like there is code missing on the generated code. 

For example instruction in `template.json`, this replaces `#define with_auth` with an empty line but I rather remove the line instead. 
```
  "with_auth_replace": {
      "type": "generated",
      "generator": "constant",
      "parameters": {
        "value": ""
      },
      "replaces": "#define with_auth"
    },
```

**Comment by YuliiaKovalova** [comment]
Hi @siganberg,

I can recommend using https://github.com/dotnet/templating/wiki/Conditions statements for fulfilling this goal.
Take a look at this sample template for getting more clarity: https://github.com/dotnet/templating/tree/main/dotnet-template-samples/content/15-computed-symbol.

**Reply by siganberg** [reply]
Not sure I explained what I'm trying to achieve. 

I'm looking for something that actually deletes the line instead of not rendering the line.

The background is I need to use this syntax `#define with_auth` on the top of my source code to actually default the value of `with_auth` to true, else if I use the `with_auth` parameters in the #if directive, it will always be evaluated to false. Default value assignment in template.json is useless there, c# conditional directive doesn't use it during design time and only runtime of the template.  

example:

```
#define with_auth 
using Microsoft.Extensions.Hosting;
#if (with_auth)
using Microsoft.AspNetCore.Hosting;
#endif
```

With the replace command the final render will be like this if with_auth is true. 

```
{empty line here} 
using Microsoft.Extensions.Hosting;
using Microsoft.AspNetCore.Hosting;
```

I can get away without using the syntax `#define with_auth` by negating it and using `no_auth` instead but this will resort to double negation and is very confusing for my user. 

example:
```
using Microsoft.Extensions.Hosting;
#if (!no_auth)
using Microsoft.AspNetCore.Hosting;
#endif
```

**Comment by YuliiaKovalova** [comment]
@siganberg, currently, Template Engine doesn't provide the described functionality.
You can create a [feature request](https://github.com/dotnet/templating/issues/new?assignees=&labels=&projects=&template=feature_request.yml) ticket in our backlog.
It will be evaluated and prioritized later.
